knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go

```diff
@@ -44,52 +40,55 @@ func Apply(instance *servingv1alpha1.KnativeServing, apiclient client.Client, sc
 	return nil
 }
 
-// reconcileKnCCDResources reconciles required resources viz Deployment, Service, Route
+// reconcileKnCCDResources reconciles required resources viz Knative Service
 // which will serve kn cross platform binaries within cluster
 func reconcileKnCCDResources(instance *servingv1alpha1.KnativeServing, apiclient client.Client, scheme *runtime.Scheme) error {
-	log.Info("Installing kn ConsoleCLIDownload resources")
-	manifest, err := manifest(instance, apiclient, scheme)
-	if err != nil {
+	if !instance.Status.IsReady() {
+		return fmt.Errorf("Knative instance %q/%q not ready yet", instance.Name, instance.GetNamespace())
+	}
+	if	err := servingv1.AddToScheme(scheme); err != nil {
 		return err
 	}
 
-	if err := manifest.Apply(); err != nil {
-		return fmt.Errorf("failed to apply kn ConsoleCLIDownload resources manifest: %w", err)
+	log.Info("Installing kn ConsoleCLIDownload resources")
+	service := &servingv1.Service{}
+	err := apiclient.Get(context.TODO(), client.ObjectKey{Namespace: instance.GetNamespace(), Name: knConsoleCLIDownloadService}, service)
+	switch {
+	case apierrors.IsNotFound(err):
+		tmpService := makeKnService(os.Getenv("IMAGE_KN_CLI_ARTIFACTS"), instance)
+		if err := apiclient.Create(context.TODO(), tmpService); err != nil {
+			return err
+		}
+	case err == nil:
+		tmpService := makeKnService(os.Getenv("IMAGE_KN_CLI_ARTIFACTS"), instance)
+		serviceFromClusterDC := service.DeepCopy()
+		changed := false
+		if !equality.Semantic.DeepEqual(service.Spec, tmpService.Spec.Template.Spec) {
```

Comment:
This will always be true I believe since you're comparing the template spec to the top-level spec. Please use the top-level spec in both cases.

Reply:
Done in https://github.com/openshift-knative/serverless-operator/pull/381/commits/f79521812b43c1d2cad86c240249dbb3633dbb16.